drivers/wallclock/sun6i-cnt64.c

```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright © 2017-2018 The Crust Firmware Authors.
+ * SPDX-License-Identifier: (BSD-3-Clause OR GPL-2.0)
+ */
+
+#include <compiler.h>
+#include <drivers/clock.h>
+#include <drivers/wallclock.h>
+#include <error.h>
+#include <mmio.h>
+#include <util.h>
+
+#define  OSC24M_CNT64_CTRL_REG 0x280
+#define OSC24M_CNT64_LOW_REG   0x284
+#define OSC24M_CNT64_HIGH_REG 0x288
+
+static uint64_t
+sunxi_wallclock_read(struct device *dev)
+{
+	uint64_t time;
+	uint32_t high_reg;
+	uint32_t low_reg;
+
+	mmio_setbits32(dev->regs + OSC24M_CNT64_CTRL_REG,
+	                                BIT(1));
+	while (mmio_read32(dev->regs + OSC24M_CNT64_CTRL_REG)) {
+		//do nothing
+	}
+
+	low_reg  = mmio_read32(dev->regs + OSC24M_CNT64_LOW_REG);
+	high_reg = mmio_read32(dev->regs + OSC24M_CNT64_HIGH_REG);
+	time     = ((uint64_t)low_reg << 32) | high_reg;
+
+	return time;
+}
+
+static int
+sunxi_wallclock_probe(struct device *dev __unused)
```

Comment:
This isn't your fault, but please move this function definition between the two structs below (we're going for topological ordering of functions).